.github/actions/build-and-test-branch/action.yml

```diff
@@ -0,0 +1,115 @@
+name: 'Build and test branch'
+description: 'Builds and tests a branch'
+inputs: 
+  branch-type:
+    description: 'String denoting either `target` or `feature` branch'
+    required: true
+  project-name:
+    description: 'String denoting the name of the project'
+    required: true
+  secrets:
+    description: 'Secrets from main.yml as JSON'
+runs:
+  using: 'composite'
+  steps:
+    - name: Install Java, GDAL, and other system dependencies
+      run: |
+        sudo apt update
+        sudo apt-get install libxml2-dev libpq-dev openjdk-8-jdk libgdal-dev libxslt-dev
+        echo Postgres and ES dependencies installed
+      shell: bash
+
+    - name: Set up Elasticsearch
+      uses: ankane/setup-elasticsearch@v1
+      with:
+        elasticsearch-version: 8
+
+    - name: Install Python packages
+      run: |
+        python -m pip install --upgrade pip
+        pip install .
+        pip install -r ${{ inputs.project-name }}/install/requirements.txt || true
```

Comment:
Can I ask why this line is necessary? I thought we wanted to encourage putting python dependencies in pyproject.toml in the usual way now that we updated the project templates. Then all you need is `pip install .` and you get everything. (AFAICT, we really only need the install directory now to pass development dependencies to pip.)

Perhaps the README's on arches applications like AfS need to clarify this now that the project templates are updated.

Reply:
btw, AfS already declares python dependencies on other django apps and arches apps (like arches_templating) in this way, and doesn't have an install dir because it doesn't have its own dev dependencies.

Reply:
I already started on #11057 :-)

Reply:
cool, I'm happy to couch this until #11057  is merged in. However I'd like like to get it in #1041 is merged, if possible I'd like test coverage to get folded into that